sunspot/lib/sunspot/session_proxy/sharding_session_proxy.rb

```diff
@@ -70,6 +70,20 @@ def index!(*objects)
         using_sharded_session(objects) { |session, group| session.index!(group) }
       end
 
+      #
+      # See Sunspot.atomic_update
+      #
+      def atomic_update(*objects)
```

Comment:
actually I marked `atomic_update` methods as `not_supported` for sharding proxies. I guess these classes were designed for pre-Solr4, where SolrCloud feature goes out-of-box